Adjust test from last PR (Version Bumps and Directory Changes)

Lalaluka opened the pull request:
Hi,
the two commits applying the Change Requests from #38

Windows compatibility:
Well the script itself runs on windows without a problem (tested with git Bash). The Problem is the Path separator ( / vs \ )in the package.json. I don't really know if there is a solution that could be applied here easily.

Lalaluka commented:
Something is failing the scripts in Actions. I'm on it

joeldenning commented on tests/testVue.sh:
```diff
@@ -1,14 +1,16 @@
+mkcdir ()
+{
+    mkdir -p -- "$1" &&
+      cd -P -- "$1"
```
Changing directories during the test command is not great, as when the tests failed the user will be confused as to why their terminal is suddenly in a new directory compared to when they started it. Instead, let's use pnpm `--dir` flag:

https://pnpm.io/pnpm-cli#-c-path---dir-path

Lalaluka replied:
While we can use `--dir` for later executions once the application is created this won't apply to vue cli.
So I'm not sure if that will improve things a lot.

`vue create --help`
Does not list a similar function like pnpm --dir

joeldenning replied:
In that case, it would be best to try to catch the error and return the user back to their original working directory, if possible.

Lalaluka replied:
It should do that by default since concurrently creates sub Processes right? The Process the Users Terminal is running in won't change.

joeldenning commented on .github/workflows/build_and_test.yml:
```diff
@@ -42,10 +42,10 @@ jobs:
+          version: 6.15.0
       - run: pnpm add -g @vue/cli
       - run: pnpm install --frozen-lockfile
       - run: pnpm run test:vue3
```
can't we just have one test stage that runs `pnpm run test` rather than a test_vue2 and test_vue3 stage?

Lalaluka replied:
Sure we can do it this way.
I thought that two sets might be more helpful to find the root of an issue.

But yea it will simplify things only having a single step.

Lalaluka replied:
Part of 6dee007

joeldenning commented on tests/testVue.sh:
```diff
@@ -1,14 +1,16 @@
 #!/bin/bash
 
-i= basename "$PWD"
+mkcdir ()
```
Please move this file to the `test` folder rather than the `tests` folder

Lalaluka replied:
Part of 6dee007

joeldenning reviewed (commented):
The error in CI seems to be occurring during the `vue invoke single-spa` command. Vue CLI runs `pnpm install --reporter silent` as shown in https://github.com/single-spa/vue-cli-plugin-single-spa/pull/39/checks?check_run_id=3678996808#step:7:35

The `--reporter silent` swallows the error which is why there isn't more useful info. Trying to find a way to get it not use the silent reporter would be the best way to debug. If we can't figure it out we can try switching to `npm` rather than `pnpm` for the vue cli projects.

Lalaluka commented:
The problem with the pnpm issue is that i cant reproduce it on either MacOS nor Windows10. I'll setup a Ubuntu VM and run some tests there.

Lalaluka commented:
Ok only took me 2 months to come back to this issue (sorry for that).

But I moved the tests over to npm (still can't get pnpm to work). Also renamed the step to just test.
Should work on Windows(with GitBash), Mac and Linux.
